modules/components/core/org.wso2.carbon.pc.core/src/main/java/org/wso2/carbon/pc/core/import_export/ProcessImport.java

```diff
@@ -0,0 +1,219 @@
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.nio.file.Files;
+import java.util.ArrayList;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+
+public class ProcessImport {
+    private static final Log log = LogFactory.getLog(ProcessImport.class);
+    File[] listOfProcessDirs;
+    RegistryService registryService = ProcessCenterServerHolder.getInstance().getRegistryService();
+    UserRegistry reg;
+    String user;
+
+    /**
+     * @param processZipInputStream
+     * @param user
+     * @throws IOException
+     * @throws RegistryException
+     * @throws ProcessCenterException
+     * @throws UserStoreException
+     */
+    public void importProcesses(InputStream processZipInputStream, String user)
+            throws IOException, RegistryException, ProcessCenterException, UserStoreException {
+
+        if (registryService != null) {
+            reg = registryService.getGovernanceUserRegistry(user);
+            this.user = user;
+            //extract zip file stream to the system disk
+            byte[] buffer = new byte[2048];
+            ZipInputStream zipInputStream = new ZipInputStream(processZipInputStream);
+            new File(ImportExportConstants.IMPORTS_DIR).mkdirs();
+
+            try {
+                ZipEntry entry;
+                while ((entry = zipInputStream.getNextEntry()) != null) {
+                    //counter++;
+                    String outpath = ImportExportConstants.IMPORTS_DIR + "/" + entry.getName();
+                    String dirPath = outpath.substring(0, outpath.lastIndexOf("/"));
+                    new File(dirPath).mkdirs();
+                    FileOutputStream fileOutputStream = null;
+                    try {
+                        fileOutputStream = new FileOutputStream(outpath);
+                        int len = 0;
+                        while ((len = zipInputStream.read(buffer)) > 0) {
+                            fileOutputStream.write(buffer, 0, len);
+                        }
+                    } finally {
+                        if (fileOutputStream != null) {
+                            fileOutputStream.close();
+                        }
+                    }
+                }
+            } finally {
+                zipInputStream.close();
+            }
+
+            //check if the importing processes are already available
+            boolean isProcessAvailableAlready = isProcessesAlreadyAvailble();
+            if (isProcessAvailableAlready) {
+                return;
+            }
+
+            //else do the process importing for each process
+            for (File processDir : listOfProcessDirs) {
+                if (processDir.isDirectory()) {
+                    String processDirName = processDir.getName();
+                    String processDirPath = processDir.getPath();
+                    String processRxtPath = processDirPath + "/" + "process_rxt.xml";
```

Comment:
Thanks @Niranjan-K. Used Java 7 Paths and fixed with a634a28e2a8d9214670984aa763dd69483641b78